app/Platform/Actions/UpdatePlayerMetrics.php

```diff
@@ -0,0 +1,49 @@
+<?php
+
+declare(strict_types=1);
+
+namespace App\Platform\Actions;
+
+use App\Site\Models\User;
+
+class UpdatePlayerMetrics
+{
+    public function execute(User $user): void
+    {
+        $playerGames = $user->playerGames()->where('achievements_unlocked', '>', 0);
+        $user->achievements_unlocked = $playerGames->sum('achievements_unlocked');
+        $user->achievements_unlocked_hardcore = $playerGames->sum('achievements_unlocked_hardcore');
+        $user->completion_percentage_average = $playerGames->average('completion_percentage');
+        $user->completion_percentage_average_hardcore = $playerGames->average('completion_percentage_hardcore');
+
+        // TODO refactor to use aggregated player_games metrics
+        $user->RAPoints = $user->achievements()->published()->wherePivotNotNull('unlocked_hardcore_at')->sum('Points');
+        $user->RASoftcorePoints = $user->achievements()->published()->wherePivotNull('unlocked_hardcore_at')->sum('Points');
+        $user->TrueRAPoints = $user->achievements()->published()->wherePivotNotNull('unlocked_hardcore_at')->sum('TrueRatio');
+
+        // TODO refactor to use the above implementation only
```

Comment:
Removed the whole block in https://github.com/RetroAchievements/RAWeb/pull/1874/commits/a9552814b9248b7cb6b2e8319c9701f3dce3ac25

I wasn't sure if we'd risk running into temporary points/rank jumps when basing the calculation on player_achievements because of known zombie entries.
As cleaning up those zombie entries before this code even hits production this shouldn't be an issue eventually.